adb/device.go

```diff
@@ -165,13 +161,14 @@ func (d *Device) Upgrade(path string) error {
 	return d.install([]string{"-r"}, path)
 }
 
-func getResultLine(out io.Reader) string {
-	scanner := bufio.NewScanner(out)
-	for scanner.Scan() {
-		l := scanner.Text()
-		if strings.HasPrefix(l, "Failure") || strings.HasPrefix(l, "Success") {
-			return l
-		}
+func getResultLine(output []byte) string {
+	str := strings.TrimRight(string(output[:]), "\n")
```

Comment:
Why `output[:]`? When you convert to string, the underlying data is already copied.

Also note that the previous method read line by line until one was found that matched what we were looking for. This is different; it will break if `adb` returns multiple lines. I've had this happen in the past with warnings and such, and it will only get worse now that we're capturing all the output. So I'd rather iterate over the lines as before.